building/gits/optee_os.rst

```diff
@@ -498,5 +498,158 @@ Some combinations of configuration variables may not be valid. This should be
+    e0186224 libmbedtls: refine mbedtls license header
+    215609ae mbedtls: configure mbedtls to reach for config
+    6916dcd9 mbedtls: remove default include/mbedtls/config.h
+    b60fc42a Import mbedtls-2.6.1
+
+Commit b60fc42a imports the library under ``lib/libmbedtls/mbedtls`` with no
+modification to the code (not the whole library is imported however, since some
+files are not needed they are deleted as mentioned in the commit description).
+Then a couple of adjustments are made in commits 6916dcd9 and 215609ae in order
+to be able to build with optee_os. At this point the initial import is done and
+subsequent commits are local improvements or bug fixes made later on.
+
+The initial import (commits b60fc42a, 6916dcd9 and 215609ae) is merged into
+master as a "squashed" commit to preserve bisectability -- in other words, so
+that no commit in master breaks the build::
+
+    $ git show --quiet 817466cb
+    commit 817466cb476de705a8e3dabe1ef165fe27a18c2f
+    Author: Jens Wiklander <jens.wiklander@linaro.org>
+    Date:   Tue May 22 13:49:31 2018 +0200
+
+        Squashed commit importing mbedtls-2.6.1 source
+    
+        Squash merging branch import/mbedtls-2.6.1
+    
+        215609ae4d8c ("mbedtls: configure mbedtls to reach for config")
+        6916dcd9b9cd ("mbedtls: remove default include/mbedtls/config.h")
+        b60fc42a5cd5 ("Import mbedtls-2.6.1")
+    
+        Acked-by: Joakim Bech <joakim.bech@linaro.org>
+        Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>
+
+Later changes are reviewed and merged into master normally, and are also
+recorded on top of the import branch via pull requests against the import
+branch. Consider for example the two following commits, one is in the master
+branch and the other is applied on top of the import branch::
+
+    18c5148d357e ("mbedtls: add mbedtls_mpi_init_static()")  # In master
+    f934e2913b7b ("mbedtls: add mbedtls_mpi_init_static()")  # In import/mbedtls-2.6.1
+
+The master branch is occasionally merged into the import branches. Otherwise,
+some patches cherry-picked from master would not apply. Note that it is a
+"normal" merge (not a rebase), so the commits that are on master can easily
+be filtered out (``git log --oneline --first-parent import/...``).
+
+When it is time to upgrade a library, a new import branch is created from
+master, for example: ``import/mbedtls-2.16.0``. A pull request is created against
+this branch with the following commits:
+
+ - The first commit deletes the "old" version of the library and imports the
+   new upstream version.
+ - The subsequent commits are cherry-picked from the previous import branch and
+   adjusted as needed. These commits are effectively "rebased" onto the new
+   library.
+ - Build files are updated if needed.
+
+Here is the history of the ``import/mbedtls-2.16.0`` branch, for comparison with
+the initial import::
+
+    $ BRANCH=github/import/mbedtls-2.16.0
```

Comment:
I would use code-blocks (bash) for things run in a shell.

Reply:
The problem is, it will apply syntax highlighting to non-bash text (the git output). I think code blocks should be used only for a "true" script. Even the prompt (`$`) is not bash syntax in fact.

![image](https://user-images.githubusercontent.com/8288195/78780977-bfa14680-799f-11ea-8147-bd5b73d8d995.png)

Reply:
Hmmm I noticed the block after "on top of the import branch" is rendered with some sort of syntx highlighting although I haven't specified any code-block / language. Why is that?

![image](https://user-images.githubusercontent.com/8288195/78781323-5241e580-79a0-11ea-94b8-74d7afc1872d.png)

Reply:
You can try with `.. code-block:: none` in that case.

Reply:
> You can try with `.. code-block:: none` in that case.

Works fine indeed. PR updated. Thanks.

Reply:
> Hmmm I noticed the block after "on top of the import branch" is rendered with some sort of syntax highlighting although I haven't specified any code-block / language. Why is that?

Sphinx or RST (not sure which) defaults to python if no explicit language is given.

Reply:
OK good to know, thanks